### PR TITLE
[SPARK-6374] [MLlib] add get for GeneralizedLinearAlgo

### PR DESCRIPTION
**Pull request by hhbyyh**
I find it's better to have getter for NumFeatures and addIntercept within GeneralizedLinearAlgorithm during actual usage, otherwise I 'll have to get the value through debug.

### CODE REVIEW

**Comment by srowen** [comment]
LGTM. It seems harmless to expose these, and similarly-named getters are exposed elsewhere.

File: mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala [hunk 1]
```diff
@@ -142,6 +147,11 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
   protected def createModel(weights: Vector, intercept: Double): M
 
   /**
+   * Get if the algorithm uses addIntercept
+   */
+  def isAddIntercept: Boolean = this.addIntercept
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: minor: `isAddIntercept` to `shouldAddIntercept`? I'm not sure which reads better. @srowen

Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: Heh, `should` is more natural although I'm used to `is` or `get` for getters in Java. I had the same reaction but think `is`is OK. It's more obviously a getter to my eyes.

**Comment by mengxr** [comment]
Merged into master. Thanks!

File: mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: Okay, let's use `is`.

**Pull request closed without merging**